### ISSUE TITLE
Uncaught Error: Loading chunk 0 failed.

### ISSUE THREAD

**Issue by georeith** [webpack/webpack#9207]
# Bug report

**What is the current behavior?**

Webpack is failing to load an externally hosted chunk. The request for the chunk responds 200 and the response is fine:

![network response codes](https://user-images.githubusercontent.com/3090066/58637674-ed7f1e80-82ea-11e9-957b-5ac7dd89d5c4.png)
![response](https://user-images.githubusercontent.com/3090066/58637695-fa9c0d80-82ea-11e9-8c80-a33e0d841bc3.png)

however webpack gives the following error:

![error](https://user-images.githubusercontent.com/3090066/58637714-0b4c8380-82eb-11e9-9383-dfc1e6bb1e6c.png)

I inspected the error and this is the state during the error triggering:

![image (20)](https://user-images.githubusercontent.com/3090066/58638034-c543ef80-82eb-11e9-9a20-14c2fc4a895a.png)

**If the current behavior is a bug, please provide the steps to reproduce.**

I am unable to share the codebase but the chunk in question is created like so:

```js
const PrototypeViewer = lazy(() =>
  import(/* webpackChunkName: "PrototypeViewer" */ '../../components/PrototypeViewer')
);
```

**What is the expected behavior?**
The chunk should be loaded without error.

**Other relevant information:**
webpack version:  4.28.2
Node.js version: 12.3.1
Operating System: Ubuntu Linux 16.04 LTS
Additional tools:

**Comment by evilebottnawi** [comment]
Network problem? Can you create reproducible test repo?

**Comment by georeith** [comment]
@evilebottnawi Working on it.

I have made some more discovery and believe it is due to the combination of `config.externals` and code splitting.

Locally I get a different (but I think related error) that my chunk is trying to load `react` despite it not being included due to my `config.externals` configuration:
```js
{
  react: 'React',
  'react-dom': 'ReactDOM',
};
```
![image](https://user-images.githubusercontent.com/3090066/58718419-5d61d780-83c5-11e9-8a35-335f64637b82.png)
![image](https://user-images.githubusercontent.com/3090066/58718448-72d70180-83c5-11e9-90dd-65fba8a9f3d5.png)

**Comment by evilebottnawi** [comment]
hm, strange

**Comment by georeith** [comment]
OK I have worked out what the issue is here.

We returned an array of configs out of `webpack.config.js`. They all wrote to the same output directory and shared some chunk names. The chunks appear to differ slightly in content and thus one of the configs would get the chunks intended for it and the other's would be pulling in the wrong ones. The solution was to add:

```js
output: {
    chunkFilename: '[name].[chunkhash].js',
}
```

Closing this as this is more of a configuration issue, not much webpack could do I believe other than warn about it.
